### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by blcts**

### CODE REVIEW

**Review by Oleksandr-Pustovit — APPROVED** [review]
Good job!

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -2,14 +2,44 @@
...
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    Animal.alive.push(this);
```
Comment (VETALEK) [inline src/herbivoresAndCarnivores.js:31]: you can move it to the `Animal`'s constructor 😉

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -2,14 +2,44 @@
...
+  }
+
+  bite(herbi) {
+    if (herbi.hidden === false && herbi instanceof Herbivore) {
+      herbi.health -= 50;
+    }
+
+    if (herbi.health <= 0) {
+      Animal.alive = Animal.alive.filter(animal => animal.name !== herbi.name);
```
Comment (VETALEK) [inline src/herbivoresAndCarnivores.js:40]: you remove all the animals with the same name 😢 
mb use `animal === herbi`?

**Review by VETALEK — CHANGES_REQUESTED** [review]
Good start 💪

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -2,14 +2,38 @@
...
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
```
Comment (VETALEK) [inline src/herbivoresAndCarnivores.js:26]: you still need
```
constructor(...args) {
  super(..args)
}
```

**Review by VETALEK — CHANGES_REQUESTED** [review]
Almost 💪

**Review by VETALEK — APPROVED** [review]
GJ 👍

**Pull request closed without merging**